subprojects/ide-native/src/main/java/org/gradle/ide/visualstudio/VisualStudioExtension.java

```diff
@@ -20,17 +20,12 @@
 import org.gradle.api.NamedDomainObjectSet;
 
 /**
- * The configuration for mapping a set of {@link org.gradle.nativeplatform.NativeComponentSpec}s to a Visual Studio project.
+ * The configuration for mapping a set of C++ components to a Visual Studio project.
  */
 @Incubating
 public interface VisualStudioExtension {
     /**
      * The {@link VisualStudioProject}s generated.
      */
     NamedDomainObjectSet<? extends VisualStudioProject> getProjects();
```

Comment:
Should we keep this a `NamedDomainObjectSet` or reduce it to a `Collection` or `Set`?  Is there a use case for plugins/builds to add their own `VisualStudioProject`s or remove/change the ones that we add automatically?

Reply:
I think it's so that you can do named configuration of individual component project files instead of using `projects.all{}`, e.g.
```
model {
  visualStudio {
    projects.helloDll {
      projectFile.location = "vs/helloSharedLibrary.vcxproj"
    }
  }
}
```